src/vaults/PoolManager.sol

```diff
@@ -210,7 +203,18 @@ contract PoolManager is Auth, Recoverable, IPoolManager, IUpdateContract, IPoolM
             IShareToken(shareToken_).file("hook", hook);
         }
 
-        pools[poolId].shareClasses[scId].shareToken = shareToken_;
+        pool.shareClasses[scId].shareToken = shareToken_;
+
+        address escrow = poolEscrowFactory.deployedPoolEscrow(poolId);
+        // Deploy new escrow only on first added share class for pool
+        if (escrow == address(0)) {
+            escrow = poolEscrowFactory.newEscrow(poolId);
+        }
+
+        // Update escrow as member to enable minting, burning and transfers on deposit and redeem
+        bytes memory updateMember =
```

Comment:
This won't apply to every hook, since some might not support the `UpdateRestrictionMember` message.

Also some might not have a hook at all (`address(0)`), for fully permissionless hooks.

I think it is just up to the pool admin to ensure this message is called. Our UI can batch it.